### ISSUE TITLE
installing idris2 api causes contrib Language.JSON module to fail

### ISSUE THREAD

**Issue by shmish111** [idris-lang/Idris2#809]
# Steps to Reproduce

```bash
make
make install-api
make install
idris2 -p contrib
> :module Language.JSON
> :exec printLn $ parse "{}"
```

# Expected Behavior
> Just {}

# Observed Behavior
> Exception in vector-ref: 1 is not a vector

# Notes
What is interesting here is that if the idris2api is not on the `IDRIS2_PATH` then this failure does not occur. Note that I called idris2 with `-p contrib` only, I didn't add `-p idris2api`.

Further I noticed that if I install the idris2api then when in an idris2 repl I _can_ load modules that are in the idris2api even though I have not added it as a package. If I do not install idris2api I cannot load these modules. So it seems idris2api modules are magically available to idris2 if they are in the `IDRIS2_PATH` where as this is not the case for other packages. In addition it seems that having these in the `IDRIS2_PATH` causes `Language.JSON` to fail.

**Comment by shmish111** [comment]
you can test this by exporting `IDRIS2_PATH` with only the package directories you want on it.

**Comment by gallais** [comment]
Both `idris2api` and `contrib` define a module `Text.Parser`.
`Language.JSON` imports transitively `Text.Parser` (via `Language.JSON.Parser`).
Could this be the source of the confusion?

**Comment by shmish111** [comment]
Do you mean 2 different modules with the same name? Well it could be but I would expect that if you don't include the package with `-p contrib` or `-p idris2api` then it shouldn't conflict?
Also, I guess this should be fixed, especially since they are both in the same repo.

**Comment by shmish111** [comment]
Indeed I can see `libs/Text/Parser.idr` and `src/Text/Parser.idr`

**Comment by andrevidela** [comment]
Should be fixed with #895
